### PR TITLE
Np 2373/add confirmation modal to action

### PR DESCRIPTION
**Pull request by RachaelDing**
https://nanopay.atlassian.net/browse/NP-2373

An example for adding confirmation model -- https://github.com/nanoPayinc/NANOPAY/pull/10853

### CODE REVIEW

**Comment by tharmaman** [comment]
@jlhughes seems like the ConfirmationModal binds to an action and in particular destructive actions. Whereas the PropertyModal binds to a property so that a user can edit a property in a separate modal (i.e. when approving an approval request so that i don't have to hit edit on a  request in order to fill things out but see the entire model). If further information for approval is needed to be store on the request, I just bind that property to the PropertyModal.

Tl;dr ConfirmationModal != PropertyModal

File: src/foam/u2/ActionView.js [hunk 1]
```diff
@@ -313,6 +318,11 @@ foam.CLASS({
         this.buttonState = undefined;
       }
     },
+    {
+      name: 'hasConfirmationModal',
+      factory: function() { 
+        return this.action && this.action.view ? true : false; }
```
Comment (tharmaman) [inline src/foam/u2/ActionView.js]:
```suggestion
        return this.action && this.action.view ? true : false;
      }
```

File: src/foam/u2/Element.js [hunk 2]
```diff
@@ -2889,18 +2889,10 @@ foam.CLASS({
...
-        return { class: 'foam.u2.ActionView', action: this };
-      }
-    }
-  ],
-
   methods: [
     function toE(args, X) {
-      var view = foam.u2.ViewSpec.createView(this.view, args, this, X);
+      var view = foam.u2.ViewSpec.createView(
+        { class: 'foam.u2.ActionView', action: this }, args, this, X);
```
Comment (tharmaman) [inline src/foam/u2/Element.js]:
```suggestion
        { class: 'foam.u2.ActionView', action: this }, args, this, X
      );
```

**Review by tharmaman — APPROVED** [review]

File: src/foam/core/Action.js [hunk 3]
```diff
@@ -110,6 +110,11 @@ foam.CLASS({
       required: true,
       value: null
     },
+    {
+      class: 'Function',
+      name: 'view',
```
Comment (kgrgreer) [inline src/foam/core/Action.js]: I think a more informative name like confirmationView would be helpful.

File: src/foam/u2/ActionView.js [hunk 4]
```diff
@@ -313,6 +318,12 @@ foam.CLASS({
         this.buttonState = undefined;
       }
     },
+    {
+      name: 'hasConfirmationModal',
+      factory: function() { 
+        return this.action && this.action.view ? true : false;
```
Comment (kgrgreer) [inline src/foam/u2/ActionView.js]: BTW, this would be simpler to just write as:
return this.action && this.action.view

But I don't think this needs to be a property, since developers aren't going to set it directly, are they? Better to make a method or just inline this logic in the one place where it is used.
